### PR TITLE
core: use InsertPoint for block inlining

### PR DESCRIPTION
**Pull request by superlopuh**
Unifies implementations

### CODE REVIEW

File: tests/test_rewriter.py [hunk 1]
```diff
@@ -147,7 +147,6 @@ def test_inline_block_at_end():
 "builtin.module"() ({
   %0 = "test.op"() : () -> !test.type<"int">
   "test.op"() ({
-  ^0:
```
Comment (superlopuh) [inline tests/test_rewriter.py:150]: inlined source blocks are now consistently deleted when inlined into another block

**Review by PapyChacal — APPROVED** [review]

**Review by compor — APPROVED** [review]

File: tests/test_rewriter.py (hunk 1 above)
Comment (math-fehr) [reply tests/test_rewriter.py:150]: Makes sense to me, that's actually what MLIR does as well!

**Review by math-fehr — APPROVED** [review]
Cool

**Review by AntonLydike — APPROVED** [review]
Very nice stuff!

**Pull request merged**